lib/plugins/requestExpiry.js

```diff
@@ -38,31 +38,45 @@ function requestExpiry(options) {
     var timeoutHeaderKey = options.timeoutHeader;
 
     return function requestExpirationCheck(req, res, next) {
-        var expiryTime;
+        /*
+         * Add check expiry API to to req if it doesn't already exist. We only
+         * add this the first time this handler is run, since this handler
+         * could be used in multiple places in the handler chain.
+         */
+        if (!req._expiry) {
+            // if the headers don't exist, then the request will never expire.
+            req._expiry = {
+                _expiryTime: Infinity,
+                isExpired: function isExpired() {
+                    return Date.now() > req._expiry._expiryTime;
+                }
+            };
 
-        if (useAbsolute) {
-            expiryTime = Number(req.headers[absoluteHeaderKey]);
-        } else {
-            // Use the start time header and add the timeout header to it
-            // to arrive at the expiration time
-            var startTime = req.headers[startHeaderKey];
-            var timeout = req.headers[timeoutHeaderKey];
+            if (useAbsolute) {
+                req._expiry._expiryTime =
+                    Number(req.headers[absoluteHeaderKey]);
+            } else {
+                // Use the start time header and add the timeout header to it
+                // to arrive at the expiration time
+                var startTime = req.headers[startHeaderKey];
```

Comment:
Same here, what if `req.headers[startHeaderKey]` or `req.headers[timeoutHeaderKey]` return NaN?  

It kind of magically works, since NaN + NaN = NaN, and any comparison against NaN always returns false (in this case, will never expire) but this seems extremely implicit. Not sure I feel super comfortable with the implicitness, though.